spec/event_store_client/adapters/grpc/commands/streams/read_spec.rb

```diff
@@ -0,0 +1,242 @@
+# frozen_string_literal: true
+
+RSpec.describe EventStoreClient::GRPC::Commands::Streams::Read do
+  let(:instance) { described_class.new }
+
+  it { is_expected.to be_a(EventStoreClient::GRPC::Commands::Command) }
+  it 'uses correct params class' do
+    expect(instance.request).to eq(EventStore::Client::Streams::ReadReq)
+  end
+  it 'uses correct service' do
+    expect(instance.service).to be_a(EventStore::Client::Streams::Streams::Stub)
+  end
+
+  describe '#call' do
+    subject do
+      instance.call(
+        stream_name,
+        options: options,
+        skip_deserialization: skip_deserialization,
+        skip_decryption: skip_decryption
+      )
+    end
+
+    let(:stream_name) { "some-stream$#{SecureRandom.uuid}" }
+    let(:options) { {} }
+    let(:skip_deserialization) { false }
+    let(:skip_decryption) { false }
+
+    context 'when stream exists' do
+      let(:event) do
+        EventStoreClient::DeserializedEvent.new(
+          id: SecureRandom.uuid, type: 'some-event', data: { foo: :bar }
+        )
+      end
+
+      before do
+        EventStoreClient.client.append_to_stream(stream_name, event)
+      end
+
+      it 'reads events from stream' do
+        is_expected.to be_a(Dry::Monads::Success)
+      end
+
+      describe 'read events' do
```

Comment:
I am testing the records in the `Dry::Monads::Success` object. If we would not be using dry-monads - then it would be not necessary, because we would test the direct result then.